modules/andes-core/client/src/main/java/org/wso2/andes/client/BasicMessageConsumer.java

```diff
@@ -619,7 +619,9 @@ public void close(boolean sendClose) throws JMSException
                         // no point otherwise as the connection will be gone
                         if (!_session.isClosed() || _session.isClosing())
                         {
-                            _session._dispatcher.rejectPending(this);
+                            if(null != _session._dispatcher) {
```

Comment:
By underlying Apache Qpid design _session._dispatcher can be null. This is because dispatcher is only started if one of following methods are called.
consumer.receive()
consumer.receiveNoWait()
consumer.setMessageListener()
Here in axis2 transport we create the message consumer to get the queue created. But during close this is thrown as here dispatcher is not set. null check is reasonable according to design as every session does not have a dispatcher.